File: src/leetcode/java/CanPlaceFlowers_605.java
package leetcode.java;

public class CanPlaceFlowers_605 {
  public boolean canPlaceFlowers(int[] flowerbed, int n) {
    boolean planted = false;
    for (int i = 0; i < flowerbed.length; i++) {
      if (flowerbed[i] == 0) {
        if (!planted) n--;
        planted = !planted;
      } else {
        if (planted) n++;
        planted = true;
      }
    }
    return n <= 0;
  }
}

File: src/leetcode/python/CanPlaceFlowers_605.py
class CanPlaceFlowers_605(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        planted = False
        for f in flowerbed:
            if f == 0:
                if not planted:
                    n -= 1
                planted = not planted
            else:
                if planted:
                    n += 1
                planted = True
        return n <= 0
